src/mds/MDSDaemon.cc

```diff
@@ -823,18 +823,19 @@ int MDSDaemon::_handle_command(
     cpu_profiler_handle_command(argvec, ds);
   } else {
     // Give MDSRank a shot at the command
-    if (mds_rank) {
+    if (!mds_rank) {
+      ss << "MDS not active";
+      r = -EINVAL;
+    }
+    else {
       bool handled = mds_rank->handle_command(cmdmap, m, &r, &ds, &ss,
 					      need_reply);
-      if (handled) {
-        goto out;
+      if (!handled) {
+        // MDSDaemon doesn't know this command
```

Comment:
@ukernel: I think here we need to clear the `ss` using `ss.str("");` or `ss.str(std::string());` here. What do you say ?

Reply:
maybe we can check if ss is empty.  if it's not, ignore appending "unrecognized command! " to it

Reply:
@ukernel: 

We could do something like:

```
if(ss.str().empty()) {
   ss << "unrecognized command! " << prefix;
}
```

But it is not necessary, as `ss` is not used anywhere before in this execution branch. Please see  _handle_command() and mds_rank->handle_command(). In this execution branch, this is the first use of `ss` after the declaration. So we could leave it as it is.

Please verify and correct me if I'm wrong :-)